### DISCUSSION TITLE
Intellisense for custom CSS properties from utils

### DISCUSSION THREAD

**Discussion by yudyananda** [stitchesjs/stitches#701] [Help] [answered]
my expectation is the custom css properties from `utils` should  have intellisense (option from appropriate theme config) , for example if it's a margin properties then should show the option from `space` config 
```javascript

const { getCssString, styled } = createCss ({
   theme: {
      space: {
        ....
      }
   },
   utils: {
      mx: config => value => ({
        marginLeft: value,
        marginRight: value,
      }),
   }
})

const Box = styled('div', {
  variants: {
    marginX: {
      auto: {
        mx: '', // should show the space option like regular margin properties
      },
    },
  },
});
```
any idea how to make intelisense working on it?

**Comment by dannytce** [comment]
Hello, afaik you need to type value. Try following:

```js
utils: {
      mx: config => (value: `$${keyof typeof config['theme']['space']}` | (string & {})) => ({
        marginLeft: value,
        marginRight: value,
      }),
}
```

**Reply by yudyananda** [reply]
how can i define this type without typescript?

**Comment by peduarte** [comment, marked as answer]
This isn't something that can magically work, but we're working on exposing types to make this easier. WIP migration guide for v1 https://stitches-site-git-v1.modulz-deploys.com/blog/migrating-from-beta-to-v1
